lib/modem_key_mgmt/modem_key_mgmt.c

```diff
@@ -128,52 +128,73 @@ int modem_key_mgmt_write(nrf_sec_tag_t sec_tag,
 	}
 
 	err = write_at_cmd_with_cme_enabled(scratch_buf, NULL, 0, &state);
-
 	return translate_error(err, state);
 }
 
-int modem_key_mgmt_read(nrf_sec_tag_t sec_tag,
-			enum modem_key_mgnt_cred_type cred_type, void *buf,
-			u16_t *len)
+/* Read the given credential into the static buffer */
+static int key_fetch(nrf_sec_tag_t tag,
+		     enum modem_key_mgnt_cred_type cred_type)
 {
 	int err;
 	int written;
+	char cmd[32];
 	enum at_cmd_state state;
 
-	if ((buf == NULL) || (len == NULL) || (*len == 0)) {
-		return -EINVAL;
-	}
-
-	written = snprintf(scratch_buf, sizeof(scratch_buf),
-			   "%s,%u,%u\r\n", MODEM_KEY_MGMT_OP_RD,
-			   (u32_t)sec_tag, (u8_t)cred_type);
+	written = snprintf(cmd, sizeof(cmd), "%s,%d,%d",
+			   MODEM_KEY_MGMT_OP_RD, tag, cred_type);
 
-	if ((written < 0) || (written >= sizeof(scratch_buf))) {
+	if (written < 0 || written >= sizeof(cmd)) {
 		return -ENOBUFS;
 	}
 
-	err = write_at_cmd_with_cme_enabled(scratch_buf, scratch_buf,
+	LOG_DBG("Sending: %s", log_strdup(cmd));
+	err = write_at_cmd_with_cme_enabled(cmd, scratch_buf,
 					    sizeof(scratch_buf), &state);
-	if (err == 0) {
-		size_t size;
-		struct at_param_list cmng_list;
+	if (err) {
+		return translate_error(err, state);
+	}
 
-		at_params_list_init(&cmng_list, AT_CMNG_PARAMS_COUNT);
-		at_parser_params_from_str(scratch_buf, NULL, &cmng_list);
-		at_params_size_get(&cmng_list, AT_CMNG_CONTENT_INDEX, &size);
+	return 0;
+}
 
-		if (size < *len) {
-			at_params_list_free(&cmng_list);
-			return -ENOMEM;
-		}
+int modem_key_mgmt_read(nrf_sec_tag_t sec_tag,
+			enum modem_key_mgnt_cred_type cred_type,
+			void *buf, u16_t *len)
+{
+	int err;
+	size_t size;
+	struct at_param_list cmng_list;
+
+	if (buf == NULL || len == NULL) {
+		return -EINVAL;
+	}
+
+	err = key_fetch(sec_tag, cred_type);
+	if (err) {
+		return err;
+	}
 
-		at_params_string_get(&cmng_list, AT_CMNG_CONTENT_INDEX, buf,
-				     &size);
-		*len = size;
+	/* Must be freed */
+	at_params_list_init(&cmng_list, AT_CMNG_PARAMS_COUNT);
+	at_parser_params_from_str(scratch_buf, NULL, &cmng_list);
+
+	at_params_size_get(&cmng_list, AT_CMNG_CONTENT_INDEX, &size);
```

Comment:
`at_params_string_get` checks if the string fits into the output buffer, ain't this enough?

Reply:
nice, thanks